source/m2mtlvserializer.cpp

```diff
@@ -126,8 +126,21 @@ bool M2MTLVSerializer::serialize_resource(const M2MResource *resource, uint8_t *
 {
     bool success = false;
     if(resource->name_id() != -1) {
-        success = serialize_TILV(TYPE_RESOURCE, resource->name_id(),
+        if ( resource->resource_instance_type() == M2MResourceInstance::INTEGER) {
+            int64_t valueInt = resource->get_value_int();
+            uint32_t buf_size;
+            /* max len 8 bytes */
```

Comment:
It might be variable in future. It depends how we going to implement support for TLV integer. In ideal case these should be supported 8, 16, 32 and 64-bit. If we are going use fixed size, e.g. 64-bit always, then we can simplify this later.

Specs says that. "Represented as a binary signed integer in network byte order, and in two‟s complement representation. The value may be 1 (8-bit), 2 (16-bit), 4 (32-bit) or 8 (64-bit) bytes long as indicated by the Length field. When transmitted over network, the data is represented in network byte order (big endian)."